### PR TITLE
Note feature

### PR DESCRIPTION
**Pull request by vishal804**
Hello
Created Notes Functionality in this
Please Review 

### Files To Review 

src/component/colorPalette/ColorPalette.js
src/component/createNote/CreateNote.js
src/component/displayNote/DisplayNote.js
src/component/sideNavbar/SideNavbar.js
src/pages/archiveNotePage/ArchiveNotePage.js
src/pages/trashNotePage/TrashNotePage.js
src/utility/NoteFunction.js

[Preview](https://deploy-preview-3--notekeep-keeper.netlify.app/)

### CODE REVIEW

File: src/component/sideNavbar/SideNavbar.js [hunk 1]
```diff
@@ -1,55 +1,72 @@
...
+            </p>
+          </li>
+          <li>
+            <p className="list-style flex flex-justify-center">
+              <Link to="/trash">
+                <i className="icon-style fas fa-trash"></i>Trash
+              </Link>
+            </p>
+          </li>
+          <li>
```
Comment (BalajiHarinath) [inline src/component/sideNavbar/SideNavbar.js:43]: Suggestion: Can use map for this repeating structure.

File: src/utility/NoteFunction.js [hunk 2]
```diff
@@ -39,3 +46,77 @@ export const updateNote = async (editNote, noteState, noteDispatch, token) => {
...
+      `/api/notes/archives/${note._id}`,
+      { note: note },
+      {
+        headers: {
+          authorization: token,
+        },
+      }
+    );
+    noteDispatch({ type: "ADD_NOTE", payload: response.data.notes });
+    noteDispatch({ type: "ARCHIVE_NOTE", payload: response.data.archives });
```
Comment (BalajiHarinath) [inline src/utility/NoteFunction.js]: Suggestion: Can check for the response status and if the promise is resolved then perform the dispatch.

File: src/utility/NoteFunction.js [hunk 3]
```diff
@@ -39,3 +46,77 @@ export const updateNote = async (editNote, noteState, noteDispatch, token) => {
...
+  try {
+    const response = await axios.delete(`/api/archives/delete/${note._id}`, {
+      headers: {
+        authorization: token,
+      },
+    });
+
+    noteDispatch({
+      type: "TRASH_ARCHIVE_NOTE",
+      payload: response.data.archives,
```
Comment (BalajiHarinath) [inline src/utility/NoteFunction.js]: Suggestion: Can check for the response status and if the promise is resolved then perform the dispatch.

File: src/utility/NoteFunction.js [hunk 4]
```diff
@@ -39,3 +46,77 @@ export const updateNote = async (editNote, noteState, noteDispatch, token) => {
...
+        },
+      }
+    );
+    noteDispatch({
+      type: "RESTORE_ARCHIVE_NOTE",
+      payload: {
+        notesData: response.data.notes,
+        archivedData: response.data.archives,
+      },
+    });
```
Comment (BalajiHarinath) [inline src/utility/NoteFunction.js]: Suggestion: Can check for the response status and if the promise is resolved then perform the dispatch.

**Review by BalajiHarinath — APPROVED** [review]
Apart from the suggested changes everything else looks good.

File: src/utility/NoteFunction.js (hunk 2 above)
Comment (vishal804) [reply src/utility/NoteFunction.js]: Done the changes

File: src/utility/NoteFunction.js (hunk 3 above)
Comment (vishal804) [reply src/utility/NoteFunction.js]: Done the changes

File: src/utility/NoteFunction.js (hunk 4 above)
Comment (vishal804) [reply src/utility/NoteFunction.js]: Done the changes

File: src/component/sideNavbar/SideNavbar.js (hunk 1 above)
Comment (vishal804) [reply src/component/sideNavbar/SideNavbar.js:43]: Will try to implement that

File: src/utility/NoteFunction.js [hunk 5]
```diff
@@ -39,3 +39,77 @@ export const updateNote = async (editNote, noteState, noteDispatch, token) => {
     console.log(error);
```
Comment (Himanshu-Dhawale) [inline src/utility/NoteFunction.js:46]: remove the console.log part

File: src/utility/NoteFunction.js [hunk 6]
```diff
@@ -39,3 +39,77 @@ export const updateNote = async (editNote, noteState, noteDispatch, token) => {
...
+      headers: {
+        authorization: token,
+      },
+    });
+    if (response.status === 200) {
+      noteDispatch({ type: "ADD_NOTE", payload: response.data.notes });
+      noteDispatch({ type: "TRASH_NOTE", payload: note });
+    }
+  } catch (error) {
+    console.log(error);
```
Comment (Himanshu-Dhawale) [inline src/utility/NoteFunction.js:62]: here too

**Review by Himanshu-Dhawale — APPROVED** [review]
Code looks clean.
All classnames are self explanatory.
just suggested a little change

**Comment by vishal804** [comment]
> Apart from the suggested changes everything else looks good.

Thanks for review

**Comment by vishal804** [comment]
> Code looks clean. All classnames are self explanatory. just suggested a little change

Thanks for review

**Pull request merged**